locale/en/LC_MESSAGES/django.po

```diff
@@ -290,53 +629,110 @@ msgstr "Maintenance"
+
+#: course/models.py
+msgid "LABEL_LATE_SUBMISSIONS_ALLOWED"
+msgstr "late submissions allowed"
+
+#: course/models.py
+msgid "LABEL_LATE_SUBMISSION_DEADLINE"
+msgstr "late submission deadline"
+
+#: course/models.py
+msgid "LABEL_LATE_SUBMISSION_PENALTY"
+msgstr "late submission penalty"
 
 #: course/models.py
 msgid "MODULE_LATE_SUBMISSION_PENALTY_HELPTEXT"
-msgstr "Multiplier of points to reduce, as decimal. 0.1 = 10%"
+msgstr "Multiplier of points to reduce as a decimal (e.g. 0.1 = 10%)."
+
+#: course/models.py
+msgid "MODEL_NAME_COURSE_MODULE"
+msgstr "course module"
+
+#: course/models.py
+msgid "MODEL_NAME_COURSE_MODULE_PLURAL"
+msgstr "course modules"
 
 #: course/models.py
 msgid "MODULE_ERROR_OPENING_TIME_AFTER_CLOSING_TIME"
-msgstr "Opening time must be earlier than the closing time."
+msgstr "The opening time must be earlier than the closing time."
 
 #: course/models.py
 msgid "MODULE_ERROR_LATE_SUBMISSION_DL_BEFORE_CLOSING_TIME"
-msgstr "Late submission deadline must be later than the closing time."
+msgstr "The late submission deadline must be later than the closing time."
 
 #: course/models.py
 msgid "MODULE_ERROR_READING_OPENING_TIME_AFTER_EXERCISE_OPENING"
 msgstr ""
-"Opening time of reading material must be earlier than the opening time of "
-"the exercises."
+"The opening time of reading material must be earlier than the opening time "
+"of the assignments."
 
 #: course/models.py
 msgid "STATUS_NO_TOTAL_POINTS"
 msgstr "No total points"
 
+#: course/models.py
+msgid "LABEL_CONFIRM_THE_LEVEL"
+msgstr "confirm the level"
+
 #: course/models.py
 msgid "LEARNING_OBJECT_CATEGORY_LEVEL_CONFIRMATION_EXERCISE_HELPTEXT"
 msgstr ""
-"Once exercise is graded non zero it confirms all the points on the hierarchy "
-"level. Implemented as a mandatory feedback feature."
+"Once an assignment is graded non-zero, it confirms all the points on the "
+"hierarchy level (e.g. within the chapter or the module). Implemented as a "
+"mandatory feedback feature."
+
+#: course/models.py
+msgid "LABEL_ACCEPT_UNOFFICIAL_SUBMITS"
+msgstr "accept unofficial submissions"
 
 #: course/models.py
 msgid "LEARNING_OBJECT_CATEGORY_ACCEPT_UNOFFICIAL_SUBMISSIONS_HELPTEXT"
 msgstr ""
-"Grade unofficial submissions after deadlines have passed or submission "
-"limits have been exceeded. The points are stored but not included in "
-"official records."
+"Accept unofficial submissions after deadlines have passed or submission "
+"limits have been exceeded. These are graded and the feedback is displayed to "
+"the student. The points are stored but not included in official records."
+
+#: course/models.py
+msgid "MODEL_NAME_LEARNING_OBJECT_CATEGORY"
+msgstr "learning object category"
+
+#: course/models.py
+msgid "MODEL_NAME_LEARNING_OBJECT_CATEGORY_PLURAL"
+msgstr "learning object categories"
 
 #: course/permissions.py
 msgid "COURSE_VISIBILITY_PERMISSION_DENIED_MSG"
-msgstr "Permission denied by course visibility"
+msgstr "Permission denied by course visibility."
```

Comment:
I'm not quite sure what the "Permission denied by course visibility" means. How can course visibility deny anything? What is the context for this?

Reply:
This appears, for example, if a user tries to access a course that isn't visible to them (e.g. an external student tries to open a course for only internal students, or a student has copy/pasted or guessed a link to course whose material is not visible to students yet.) The error messages can be something like this:
![image](https://user-images.githubusercontent.com/50318434/130032932-7a7f3d53-3a1b-4452-aa2b-4b3e8f1afd77.png)
![image](https://user-images.githubusercontent.com/50318434/130033094-69b45673-3be4-468e-976b-350e5358fa2a.png)
![image](https://user-images.githubusercontent.com/50318434/130033317-9e63d5d4-d0d5-47ef-b291-4fa266398a42.png)


I agree that the wording isn't great, but didn't have better ideas, either.

Reply:
Yes, this message definitely needs changing. It’s a very visible part of A+ that people often encounter (I surmise). 

As far as I can tell, the point of the message is that a page is not accessible, at least not now. That is, the message’s point is not what technical entity might be "denying" or "forbidding" something, so that can be left out entirely. 

Maybe something along these lines? 

If the page is unavailable to all: _“The page at this address is not available at this time. The course staff may choose to make it public later. Please also check that you are attempting to access the correct page.”_

If the page is unavailable to external students: _“Sorry! The page at this address is only available to students with an Aalto University user account. If you have an Aalto account, please make sure you use it to log in to A+.”_

Also, I’m not sure **Forbidden** in big letters is how we want to prefix this message to would-be learners.

(BTW, Ella’s message above referred to Aalto specifically, so I take it that that’s OK in this context. Or should this message be more generic?)

Reply:
>(BTW, Ella’s message above referred to Aalto specifically, so I take it that that’s OK in this context. Or should this message be more generic?)

The message uses the institution name, and I really hope that all of the institutions have names that the genetive form is formed by just adding an "n" to the end (e.g. yliopisto -> yliopiston; koulu -> koulun; lukio -> lukion), since I didn't find a way to use genetive forms properly.

Reply:
that assumption (genetive adding n) won't always work. Since languages are different, I think the only way is to have separate strings for different forms of the word. You also shouldn't manipulate any translated string by adding something to it in the code (like adding "n" to a translated string "koulu"). That won't work in all languages since they present things so differently.

Reply:
Yep, it doesn't work in all languages and would require different forms of the word. I originally didn't intend to use the Finnish brand name in the Finnish message at all, but localization and lazy formatting didn't allow not including the string in the message. Likewise having different forms of the word wouldn't work since in English here one would just say "students within {name}" while in Finnish one would need the genetive form, and providing both forms doesn't work since one would need to be left out anyway. It's a shame that localization files and formatting can't work in a way where you could provide a word with several different cases and you can just select the proper one in the translation string. I would love to just be able to write, say "Tämä kurssi on vain {institution.genetive} opiskelijoille (eli sisäisille käyttäjille)." But yep probably here it's necessary to remove the name of the institution entirely from the error message from both languages, unless it's reworded in a much longer fashion where it would be ok to say "oppilaitoksen {institution} opiskelijoille" or something like that.

Reply:
The general "forbidden" -message etc. is something that probably should be addressed in a later issue and commit. The error messages could also be improved a lot at that time. Currently with the time limits I'll change the messages to something a bit "nicer" and hopefully easier to understand ("Unfortunately you are not permitted to see this ..."), but more detailed error messages based on the context would be a good improvement that probably take too much time to fix for this RC.
I'll also change the genetive form of the institution to "oppilaitoksen {institution} opiskelijoille", etc.

Reply:
I created an issue about the access denied messages and the 403 page: #858 
If @jsorva you want to write more there, feel free to :)